### PR TITLE
CI: add autobump workflow

### PR DESCRIPTION
**Pull request by SMillerDev**
Not sure which casks to add here, but this should make sure maintainers can review and casks are updated by non-maintainers or bots

### CODE REVIEW

**Comment by reitermarkus** [comment]
> Not sure which casks to add here

I have been meaning to refactor `bump-unversioned-casks` (or reuse the logic it contains) so we don't have to hard-code this and instead just livecheck some amount of random casks that haven't been checked within the last 24 hours (or whatever timeframe).

**Review by reitermarkus — APPROVED** [review]

**Comment by reitermarkus** [comment]
> this should make sure maintainers can review and casks are updated by non-maintainers or bots

What exactly do you mean here? Make sure that PRs can be merged by a maintainer since it is opened by @BrewTestBot?

**Comment by reitermarkus** [comment]
Does `Homebrew/actions/bump-packages` run `audit` before opening a PR?

**Comment by SMillerDev** [comment]
> What exactly do you mean here? Make sure that PRs can be merged by a maintainer since it is opened by @BrewTestBot?

Exactly that. Freeing up maintainer time to do maintainer specific tasks

**Comment by reitermarkus** [comment]
> It does `brew bump` so yes

I feel like since this is automated, it should just blind-submit, since audit will need to run on PRs anyways and the cask can then be fixed in that PR.

File: .github/workflows/autobump.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Configure Git user
+        uses: Homebrew/actions/git-user-config@master
+        with:
+          username: ${{ (github.event_name == 'workflow_dispatch' && github.actor) || 'BrewTestBot' }}
+
+      - name: Bump casks
+        uses: Homebrew/actions/bump-packages@master
+        continue-on-error: true
```
Comment (reitermarkus) [inline .github/workflows/autobump.yml:37]: Is this needed? There is no step after this.

**Review by ankit-gautam23 — COMMENTED** [review]
LGTM

File: .github/workflows/autobump.yml (hunk 1 above)
Comment (SMillerDev) [reply .github/workflows/autobump.yml:37]: It's looping so I think it's needed for that.

**Comment by SMillerDev** [comment]
> I feel like since this is automated, it should just blind-submit, since audit will need to run on PRs anyways and the cask can then be fixed in that PR.

I don't think we should be autobumping broken packages. If it doesn't pass audit it should not be on the autobump list IYAM.

**Comment by reitermarkus** [comment]
> I don't think we should be autobumping broken packages.

I disagree, since broken casks will stay broken that way instead of us being able to fix them or remove them.

File: .github/workflows/autobump.yml (hunk 1 above)
Comment (reitermarkus) [reply .github/workflows/autobump.yml:37]: What do you mean looping? Steps can't loop, can they?

**Comment by reitermarkus** [comment]
Obviously in it's current opt-in state it wouldn't make sense to add broken casks in the first place, though.

**Comment by SMillerDev** [comment]
I don't think detecting issues should be the task of the autobump flow. There is an "online check" in core that can help there, it just tests a predefined amount of formula for common issues.

**Comment by reitermarkus** [comment]
> I don't think detecting issues should be the task of the autobump flow. 

I mean it would just be bumping the version but not running audit, not doing anything specific with regards to issue detection. That way simple issues like a new version having a  slightly different `.app` name can simply be fixed in a PR rather than having to submit a bump with a fix manually.

**Comment by razvanazamfirei** [comment]
I agree with @reitermarkus that it's useful to have casks w/ failed audits included, particularly if we abdicated day-to-day responsibility for the casks.

How large do we want this list to be/what are the criteria for inclusion? Do we want popular apps? Apps with frequent version updates or a very predictable livecheck and release cycle? Shouldn't apps on this list just be merged automatically if the audit passes?

**Comment by SMillerDev** [comment]
We definitely don't want automerging since a bot can't determine the impact of a change. The requirements for the list in core are essentially: will this cause problems if it gets bumped without user involvement. If it doesn't, it can go on the list.

The goal of this is in no way to replace users submitting updates, but if maintainers always go through the same list of packages to upgrade, we might as well automate it so that the maintainers can do the thing they are most important for: reviews

**Comment by MikeMcQuaid** [comment]
> If it doesn't pass audit it should not be on the autobump list IYAM.

I agree. Honestly, like homebrew/core, there shouldn't be any casks failing (non-online) audits at all IMO. If there are: let's make those "strict audits" instead.

**Pull request merged**